Tests/SafeChildProcessHandleTests.cs

```diff
@@ -211,6 +215,94 @@ public void ChildProcess_InheritsParentEnvVars_WhenEnvironmentAccessed()
         }
     }
 
+#if WINDOWS || LINUX
+    [Fact]
+#endif
```

Comment:
This test needs to ensure the env var is always used with the right value and it respects updates. No matter the OS and optimizations.

```suggestion
    [Fact]
```